src/datascience-ui/ipywidgets/renderer/index.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { NotebookOutputEventParams, NotebookRendererApi } from 'vscode-notebook-renderer';
+const JupyterIPyWidgetNotebookRenderer = 'jupyter-ipywidget-renderer';
+
+// tslint:disable: no-any no-console
+function renderOutput(e: NotebookOutputEventParams) {
+    const renderOutputFunc =
+        (window as any).ipywidgetsKernel?.renderOutput || (global as any).ipywidgetsKernel?.renderOutput;
+    if (renderOutputFunc) {
+        return renderOutputFunc(e);
+    }
+    console.error('Rendering widgets on notebook open is not supported.');
+}
+
+function disposeOutput(e: { outputId: string } | undefined) {
+    const disposeOutputFunc =
+        (window as any).ipywidgetsKernel?.disposeOutput || (global as any).ipywidgetsKernel?.disposeOutput;
```

Comment:
We don't need `global as any`.
In browser context `global === window`.

Reply:
Sorry but we do. Otherwise it doesn't work for some reason.